### PR TITLE
[WIP] Hand of god - Fully writen from zero

### PR DESCRIPTION
**Pull request by ghost**
No, i am not just taking ancient HoG code and showing it as a new gamemode. All gamemode code will be fully writen by me, and from original gamemode it takes only it's idea and some sprites. 

# What the fucck

There are some cult teams. One team(3 cultists and 1 god) per 16 people. Different teams get different colors. Every cultists needs to serve it's god, and every god needs to break free into mortal dimension to win.  Gods play in a RTS style, where they need to summon structures and make upgrades for cult equipment, and cultist play in a... well... standart cult gameplay with some RPG mechanics.

# Structures

Every god can construct various structures to guide it's cult in battle. They can be placed mostly on cult titles, or in nexus/shrine range(see further). Also god can interact structures by buffing them, or setting them into different modes.

Nexus - so, it is the main cult structure. God needs to place it before doing shit. It generates and stores cult energy. The god can recall everyone to it one time in the game. If it dies, it drops a nexus crystal, that can be sacrificed to boost another cult energy regeneration and maximum energy.

Conversion altar - place there a human with a soul, and gain a brainwashed cultist with for a small amount of cult energy.

Sacrificement altar - place there a human with a soul to sacrifice him. He will loose his sould, 20 max hp, and he will be deconverted from bloodcult/hog cult/clockcult and sent to Brazil. You will get a small pernament energy regeneration buff, maximum energy buff, and some cult energy.

Shrine - it can be placed once per area. It stores and generates energy while not destroyed, and allows to construct structures near it. It can be constructed even if not in the nexus range, but needs to be placed on a cult title.

Gate - cultists can globaly talk with eachother with it, and teleport to other gates. Can be disabled by the cult's god.

Celestial workshop - can make cult shields, cult swords and cult robes. Costs energy to use. Can be disabled by it's god.

Celestial forge - contains upgrades for cult weapons, armor and books, that can be researched by it's god.

Library - can make cult books from paper, and research advanced spells.

Fountain - produces godblood, that heals cultists, purges holy water, deconverts enemy cultists and deals them some stamina damage. Damages a bit everyone else.

Prismatic lance - SHOOTS EVERY CONSINIUS ENEMY THAT IS IN IT RANGE BY LASERS THAT DEAL 15 DAMAGE, DEALS MORE DAMAGE TO MECHAS, AND ALSO SHOOTS ENEMY STRUCTURES. 

# Cult book

It is a empower rune + clockwork slab + stunbatton + welder + handcuffs.

It can prepare spells for it's owner. Also it can do many things on hit.

Help intent - transfers energy to allies or allied structures. Can convert structures into their cult version.

Harm intent - 13 damage to enemies.

Disarm - knocks target unconsinous if it has more then 80 stamina damage, otherwise it deals 46 stamina damage, mutes, and deals 46 stamina damage again after 2 seconds. Needs to be charged to do that. If the target is an enemy cultist, mindshielded or book doesn't have charge it works like on harm intent.

Grab intent - handcuffs the target if you have energy. Not to be confused with book charge.

# God

You can break lights, convert objects in range into cult versions, emag open doors, call shuttle, delay shuttle arival, temporary disable APC's, purge holy water from cultists, recall one cultist to the nexus per 7 minutes, mass recall, prepare spells for cultists, shield structures(temporary increase their armor), overcharge structures(if it is a prismatic lance it temporary gains x2 firing speed, if not - it temporary gains a prismatic lance weapon), AND MAKE MANY OTHER SHIT FUCK I AM SO TIRED OF WRITING THIS TEXT(Everything for energy cost!!!!!!!!).

# Break free

To do this you need to:
1. Complete your cult objective(one of: sacrifice X people, convert X people, have X shrines)
2. Sacrifice a nexus crystal from other god's grave.

If your god trigers this... it will be the time for your last stand. All cultists start glowing with their cult color, GAMMA SECURITY CODE FUCK FUCK FUCK, and there will be an announcement about your nexus location, so expect all enemy cultists and station crew rushing your base. After 8 minutes your god will spawn. He... well... can gib people, instantly convert people and convert objects. He has 10000 HP, so he is very strong. But not immortal! Other gods deal bonus damage to him, and enemy cult weapons also deal more damage to him. But if your god manages to break free and survive - you all greentext

# Counterplay

EMP's damage structures, damage shields, drain energy from culties.

Nullrod can pry open cult doors and ALSO drain energy from cultists.

Haha funny gygax goes brrrrr.

Just wait untill cults cill eachother.

Explode the station nuke and die in a glorious death with your cult enemies.

(Holywater deconverts cultists)

# Why it is cool

Because it allows STRATEGICAL GAMEPLAY, POLITICS, AND TEAMWORK. MANY TEAMWORK.

Make an allience with station security to kill other cultists.

Make a cool stronghold, and laugh as enemy attacks die in holy fire.

Rush enemy bases with your prismatic lances and burn their structures down.

Get a maxcap to your nexus and die in a horrible death.

# Wiki Documentation

Document this shit lol.

# Changelog

:cl:  
wip: New HoG gamemode 
/:cl:

### CODE REVIEW

**Comment by TymurShatillo** [comment]
Sex

File: code/modules/shuttle/computer.dm [hunk 1]
```diff
@@ -137,3 +137,7 @@
 /obj/machinery/computer/shuttle/connect_to_shuttle(obj/docking_port/mobile/port, obj/docking_port/stationary/dock, idnum, override=FALSE)
 	if(port && (shuttleId == initial(shuttleId) || override))
 		shuttleId = port.id
+
+/obj/machinery/computer/shuttle/AltClick(user)
```
Comment (Bibby0110) [inline code/modules/shuttle/computer.dm]: undocumented PR

Comment (ghost) [reply code/modules/shuttle/computer.dm]: Github moment

File: code/modules/food_and_drinks/food/snacks/meat.dm [hunk 2]
```diff
@@ -340,18 +340,6 @@
 	icon_state = "shadowmeat"
 	desc = "It is covered in a strange darkness. This slab's magical properties appear to be drastically weakened due to the synthetic nature of the meat."
 
-/obj/item/reagent_containers/food/snacks/meat/slab/plagued
```
Comment (ghost) [inline code/modules/food_and_drinks/food/snacks/meat.dm]: Fuck github moment

**Comment by JamieD1** [comment]
If you plan on making a gamemode, maybe take the time to learn how to use visual studio code so you can actually test compiling instead of doing everything in web edits. Like holy shit how did you do 250 web edits in another PR.

**Comment by Altoids1** [comment]
> If you plan on making a gamemode, maybe take the time to learn how to use visual studio code so you can actually test compiling instead of doing everything in web edits. Like holy shit how did you do 250 web edits in another PR.

I mean, in the end we can have this doofus just squash it all down into a one or two commits, if you're worried about polluting the commit history with garbage.

Doing everything from webedit is absolutely bonkers but there's strictly nothing stopping our devs from doing that.

**Comment by ilikesaying** [comment]
> EMP's damage structures, damage shields, drain energy from culties.

what stops a man from spamming emp reaction?

**Comment by ghost** [comment]
> If you plan on making a gamemode, maybe take the time to learn how to use visual studio code so you can actually test compiling instead of doing everything in web edits. Like holy shit how did you do 250 web edits in another PR.

I know how to use code editors, but I don't have a stable PC access so i can't use them. So I don't have any alternatives.

**Comment by ghost** [comment]
> what stops a man from spamming emp reaction?

Getting shredded with a sword

File: yogstation.dme [hunk 3]
```diff
@@ -3524,6 +3524,15 @@
 #include "yogstation\code\modules\antagonists\gang\gang.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_datums.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_hud.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_mob.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\_hog_item.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\prismatic_lance.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structure\god_interaction\_god_action.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structure\_hog_structure.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structure\lance.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\colors.dm
```
Comment (ghost) [inline yogstation.dme]:
```suggestion
#include "yogstation\code\modules\antagonists\hand_of_god\colors.dm"
```

**Comment by ghost** [comment]
Fuck

**Comment by ghost** [comment]
Hand of Science

File: yogstation.dme [hunk 4]
```diff
@@ -3525,6 +3527,15 @@
 #include "yogstation\code\modules\antagonists\gang\gang.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_datums.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_hud.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_mob.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\_hog_item.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\prismatic_lance.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\god_action.dm"
```
Comment (ghost) [inline yogstation.dme]:
```suggestion
#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\god_action.dm"
#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\overcharge.dm"
```

File: yogstation.dme [hunk 5]
```diff
@@ -3525,6 +3527,16 @@
 #include "yogstation\code\modules\antagonists\gang\gang.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_datums.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_hud.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_mob.dm"
```
Comment (ghost) [inline yogstation.dme:3587]:
```suggestion
#include "yogstation\code\modules\antagonists\hand_of_god\god\god_mob.dm"
#include "yogstation\code\modules\antagonists\hand_of_god\god\misc_interactions.dm"
```

File: yogstation.dme [hunk 6]
```diff
@@ -3525,6 +3527,17 @@
...
+#include "yogstation\code\modules\antagonists\hand_of_god\god\misc_interactions.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\_hog_item.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\prismatic_lance.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\god_action.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\overcharge.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\_hog_structure.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\lance.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\colors.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\cult_team.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\hog_servant_datum.dm"
```
Comment (ghost) [inline yogstation.dme:3620]:
```suggestion
#include "yogstation\code\modules\antagonists\hand_of_god\hog_servant_datum.dm"
#include "yogstation\code\modules\antagonists\hand_of_god\hud.dm"
```

**Comment by ghost** [comment]
I'm impressed, this seems like a very well made gamemode. can't wait to play it!!

**Comment by ghost** [comment]
Hope i will not fuck up with teams and antag datum

File: yogstation.dme [hunk 7]
```diff
@@ -3525,6 +3527,18 @@
 #include "yogstation\code\modules\antagonists\gang\gang.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_datums.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_hud.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_mob.dm"
```
Comment (ghost) [inline yogstation.dme:3587]:
```suggestion
#include "yogstation\code\modules\antagonists\hand_of_god\god\god_datum.dm"
#include "yogstation\code\modules\antagonists\hand_of_god\god\god_mob.dm"
```

File: yogstation.dme [hunk 8]
```diff
@@ -3525,6 +3527,19 @@
...
 #include "yogstation\code\modules\antagonists\gang\gang_hud.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_datum.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_mob.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\misc_interactions.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\_hog_item.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\prismatic_lance.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\god_action.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\overcharge.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\_hog_structure.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\lance.dm"
```
Comment (ghost) [inline yogstation.dme:3612]:
```suggestion
#include "yogstation\code\modules\antagonists\hand_of_god\structures\lance.dm"
#include "yogstation\code\modules\antagonists\hand_of_god\structures\nexus.dm"
```

File: yogstation.dme [hunk 9]
```diff
@@ -3525,6 +3527,20 @@
 #include "yogstation\code\modules\antagonists\gang\gang.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_datums.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_hud.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_datum.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_mob.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\misc_interactions.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\_hog_item.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\prismatic_lance.dm"
```
Comment (ghost) [inline yogstation.dme:3595]:
```suggestion
#include "yogstation\code\modules\antagonists\hand_of_god\items\cult_book.dm"
#include "yogstation\code\modules\antagonists\hand_of_god\items\prismatic_lance.dm"
```

File: yogstation.dme [hunk 10]
```diff
@@ -3525,6 +3527,21 @@
 #include "yogstation\code\modules\antagonists\gang\gang.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_datums.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_hud.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_datum.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_mob.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\misc_interactions.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\_hog_item.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\cult_book.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\prismatic_lance.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\god_action.dm"
```
Comment (ghost) [inline yogstation.dme]:
```suggestion
#include "yogstation\code\modules\antagonists\hand_of_god\spells\spell.dm"
#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\god_action.dm"
```

File: yogstation.dme [hunk 11]
```diff
@@ -3525,6 +3527,22 @@
 #include "yogstation\code\modules\antagonists\gang\gang.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_datums.dm"
 #include "yogstation\code\modules\antagonists\gang\gang_hud.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_datum.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_mob.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\misc_interactions.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\_hog_item.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\cult_book.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\prismatic_lance.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\spells\spell.dm"
```
Comment (ghost) [inline yogstation.dme]:
```suggestion
#include "yogstation\code\modules\antagonists\hand_of_god\spells\_spell.dm"
#include "yogstation\code\modules\antagonists\hand_of_god\spells\magic.dm"
```

File: yogstation.dme [hunk 12]
```diff
@@ -3525,6 +3527,23 @@
...
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_datum.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\god_mob.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\god\misc_interactions.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\_hog_item.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\cult_book.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\items\prismatic_lance.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\spells\_spell.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\spells\magic.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\god_action.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\overcharge.dm"
```
Comment (ghost) [inline yogstation.dme:3602]:
```suggestion
#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\overcharge.dm"
#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\targeted_interactions.dm"
```

File: yogstation.dme [hunk 13]
```diff
@@ -3537,6 +3539,24 @@
...
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\god_interaction\targeted_interactions.dm"
+#include "yogstation\code\modules\antagonists\hand_of_god\structures\_hog_structure.dm"
```
Comment (ghost) [inline yogstation.dme:3607]:
```suggestion
#include "yogstation\code\modules\antagonists\hand_of_god\structures\_hog_structure.dm"
#include "yogstation\code\modules\antagonists\hand_of_god\structures\fountain.dm"
```

**Comment by ghost** [comment]
Fuck

File: code/game/machinery/doors/airlock_types.dm [hunk 14]
```diff
@@ -666,10 +671,137 @@
...
+	if(cultie)
+		if(cultie.cult == src.cult)
+			. += "It belongs to your cult."
+		else
+			. += "It belongs to a heretical cult!"
+
+/obj/machinery/door/airlock/hog/emp_act(severity)
+	if(prob(80/severity))
+		open()
+	take_damage(HOG_EMP_DAMAGE_MULTIPLER, BURN, MELEE, "sound/items/welder.ogg" , get_dir(src, src), severity)   ///HoG structures don't like EMP's.
```
Comment (Bibby0110) [inline code/game/machinery/doors/airlock_types.dm]: arguments don't make sense

Comment (ghost) [reply code/game/machinery/doors/airlock_types.dm]: What do you mean?

**Comment by TymurShatillo** [comment]
And you need to include the prite images in the description

**Comment by KittyNoodle** [comment]
absolute chad webediting a gamemode

**Comment by ghost** [comment]
Yes

File: code/game/machinery/doors/airlock_types.dm [hunk 15]
```diff
@@ -666,10 +671,137 @@
...
+	var/icon_originalname = "airlock"
+
+/obj/machinery/door/airlock/hog/examine(mob/user)
+	. = ..()
+	var/datum/antagonist/hog/cultie = IS_HOG_CULTIST(user)
+	if(cultie)
+		if(cultie.cult == src.cult)
+			. += "It belongs to your cult."
+		else
+			. += "It belongs to a heretical cult!"
```
Comment (tattax) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
	if(cultie?.cult == src.cult)
		. += "It belongs to your cult."
	else
		. += "It belongs to a heretical cult!"
```

File: code/game/machinery/doors/airlock_types.dm [hunk 16]
```diff
@@ -666,10 +671,137 @@
...
+	if(prob(80/severity))
+		open()
+	take_damage(HOG_EMP_DAMAGE_MULTIPLER*severity, BURN, MELEE, "sound/items/welder.ogg" , get_dir(src, src), severity)   ///HoG structures don't like EMP's.
+
+/obj/machinery/door/airlock/hog/canAIControl(mob/user)
+	var/datum/antagonist/hog/cultie = IS_HOG_CULTIST(user)    ///Idk maybe i will in far future make AI and silicon convertion
+	if(!cultie)
+		return FALSE
+	if(cultie.cult != src.cult)
+		return FALSE
```
Comment (tattax) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
	if(cultie?.cult != src.cult)
		return FALSE
```

File: code/game/machinery/doors/airlock_types.dm [hunk 17]
```diff
@@ -666,10 +671,137 @@
...
+		return 
+	..()
+
+/obj/machinery/door/airlock/hog/allowed(mob/M)
+	var/datum/antagonist/hog/cultie = IS_HOG_CULTIST(M)    ///Idk maybe i will in far future make AI and silicon convertion
+	if(!cultie)
+		return 0
+	if(cultie.cult != src.cult)
+		return 0
+	return 1
```
Comment (tattax) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
	if(cultie?.cult != src.cult)
		return FALSE
	return TRUE
```

File: code/game/machinery/doors/airlock_types.dm [hunk 18]
```diff
@@ -666,10 +671,137 @@
...
+
+/obj/machinery/door/airlock/hog/attackby(obj/item/I, mob/living/user, params)
+	var/datum/antagonist/hog/cultie = IS_HOG_CULTIST(user)    ///Idk maybe i will in far future make AI and silicon convertion
+	if(!cultie)
+		return ..()
+	if(cultie.cult != src.cult)
+		return ..()
```
Comment (tattax) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
	if(cultie?.cult != src.cult)
		return ..()
```

File: code/game/mecha/mecha.dm [hunk 19]
```diff
@@ -1230,3 +1230,16 @@ GLOBAL_VAR_INIT(year_integer, text2num(year)) // = 2013???
...
 	return FALSE
+
+///////////////////////
+///// Random shit /////
+///////////////////////
+
+/obj/mecha/attack_god(mob/camera/hog_god/god, modifier)
+	if(occupant)
+		var/datum/antagonist/hog/cultie = IS_HOG_CULTIST(occupant)
+		if(cultie && cultie.cult == god.cult)
```
Comment (tattax) [inline code/game/mecha/mecha.dm]:
```suggestion
		if(cultie?.cult == god.cult)
```

File: code/game/turfs/simulated/floor/misc_floor.dm [hunk 20]
```diff
@@ -290,3 +296,106 @@
...
+
+/turf/open/floor/hog/process()
+	if(!healdudes())
+		STOP_PROCESSING(SSobj, src)
+
+/turf/open/floor/hog/proc/healdudes()
+	for(var/mob/living/L in src)
+		if(L.stat == DEAD)
+			continue
+		. = 1
```
Comment (tattax) [inline code/game/turfs/simulated/floor/misc_floor.dm]:
```suggestion
		. = TRUE
```

File: code/game/turfs/simulated/floor/misc_floor.dm [hunk 21]
```diff
@@ -290,3 +296,106 @@
...
+		var/datum/antagonist/hog/cultie = IS_HOG_CULTIST(L)
+		if(!cultie || cultie.cult != src.cult)
```
Comment (tattax) [inline code/game/turfs/simulated/floor/misc_floor.dm]:
```suggestion
		if(cultie?.cult != src.cult)
```

File: code/game/turfs/simulated/floor/misc_floor.dm [hunk 22]
```diff
@@ -290,3 +296,106 @@
...
+				viewing += M.client
+		flick_overlay(I, viewing, 8)
+		L.heal_bodypart_damage(1)
+
+/turf/open/floor/hog/try_replace_tile(obj/item/stack/tile/T, mob/user, params)
+	return
+
+/turf/open/floor/hog/crowbar_act(mob/living/user, obj/item/I)
+	user.visible_message(span_notice("[user] begins slowly prying up [src]..."), span_notice("You begin painstakingly prying up [src]..."))
+	if(I.use_tool(src, user, 70, volume=80))
```
Comment (tattax) [inline code/game/turfs/simulated/floor/misc_floor.dm]:
```suggestion
	if(I.use_tool(src, user, 7 SECONDS, volume=80))
```

File: code/game/turfs/simulated/floor/misc_floor.dm [hunk 23]
```diff
@@ -290,3 +296,106 @@
...
+		user.visible_message(span_notice("[user] pries up [src], but then it vanishes in a spark of magic!"), span_notice("You pry up [src], but then it vanishes in a spark of magic!"))
+		qdel(src)
+	return TRUE
+
+/turf/open/floor/hog/proc/change_hog_team(datum/team/hog_cult/new_cult)
+	if(cult)
+		cult.objects -= src
```
Comment (tattax) [inline code/game/turfs/simulated/floor/misc_floor.dm]:
```suggestion
	cult?.objects -= src
```

File: yogstation/code/modules/antagonists/hand_of_god/actions.dm [hunk 24]
```diff
@@ -0,0 +1,22 @@
+/datum/action/innate/pray
+	name = "Talk to your God"
+	button_icon_state = "godspeak"
+	
+/datum/action/innate/pray/IsAvailable()
+	if(..())
+		if(IS_HOG_CULTIST(owner))
+			return 1
+		return 0
```
Comment (tattax) [inline yogstation/code/modules/antagonists/hand_of_god/actions.dm]:
```suggestion
	if(..() && (IS_HOG_CULTIST(owner))
		return TRUE
	return FALSE
```

File: yogstation/code/modules/antagonists/hand_of_god/actions.dm [hunk 25]
```diff
@@ -0,0 +1,22 @@
...
+	if(!cultie)
+		return
+	if(!cultie.cult.god || !cultie.cult)
+		to_chat(owner, span_warning("You don't have any god to pray to!"))
+		return
+	var/msg = input(owner,"Speak to your god","pray","") as null|text
+	if(!msg)
+		return
+	to_chat(owner, span_notice("You pray to your god: [msg]"))
+	to_chat(cultie.cult.god, span_cult("[owner] prays to you: [msg]"))
```
Comment (tattax) [inline yogstation/code/modules/antagonists/hand_of_god/actions.dm]:
```suggestion
/datum/action/innate/pray/Activate()
	var/datum/antagonist/hog/cultie = IS_HOG_CULTIST(owner)
	if(!cultie?.cult?.god)
		to_chat(owner, span_warning("You don't have any god to pray to!"))
		return
	var/msg = input(owner,"Speak to your god","pray","") as null|text
	if(!msg)
		return
	to_chat(owner, span_notice("You pray to your god: [msg]"))
	to_chat(cultie.cult.god, span_cult("[owner] prays to you: [msg]"))
```

File: yogstation/code/modules/antagonists/hand_of_god/cult_team.dm [hunk 26]
```diff
@@ -0,0 +1,128 @@
...
+		src.possible_buildings += buildin
+	
+/datum/team/hog_cult/proc/here_comes_the_money()
+	var/income = permanent_regen
+	income += energy_regen
+	for(var/obj/structure/destructible/hog_structure/shrine/shrine in objects)
+		if(!shrine)
+			continue
+		if(shrine.cult != src)
+			continue
```
Comment (tattax) [inline yogstation/code/modules/antagonists/hand_of_god/cult_team.dm]:
```suggestion
		if(shrine?.cult != src)
			continue
```

File: yogstation/code/modules/antagonists/hand_of_god/cult_team.dm [hunk 27]
```diff
@@ -0,0 +1,128 @@
...
+			if(istype(O, E))
+				apply_research_effects(O)
+
+/datum/hog_research/proc/apply_research_effects(var/obj/O)	
+	var/obj/item/hog_item/upgradeable/item = O
+	if(!item)
+		return
+	if(item.cult != cult)
+		return
+	item.upgrades += 1
```
Comment (tattax) [inline yogstation/code/modules/antagonists/hand_of_god/cult_team.dm]:
```suggestion
	if(item?.cult != cult)
		return
	item.upgrades++
```

File: yogstation/code/modules/antagonists/hand_of_god/cult_team.dm [hunk 28]
```diff
@@ -0,0 +1,128 @@
...
+	item.force = initial(item.force) += (item.upgrades * item.force_add)
+	item.throwforce = initial(item.throwforce) += (item.upgrades * item.throwforce_add)
```
Comment (tattax) [inline yogstation/code/modules/antagonists/hand_of_god/cult_team.dm]:
```suggestion
	item.force = initial(item.force) + (item.upgrades * item.force_add)
	item.throwforce = initial(item.throwforce) + (item.upgrades * item.throwforce_add)
```

**Comment by Cartlord** [comment]
I just realized, what are people who aren't part of the cults meant to DO in this mode?

**Comment by ghost** [comment]
> I just realized, what are people who aren't part of the cults meant to DO in this mode?

Uhhhh, fight them?

**Comment by boodaliboo** [comment]
reduce cult book damage 13 to strong

**Comment by ghost** [comment]
Ok

**Comment by ghost** [comment]
TODO:
1. Cut fucking away all content that doesn't have an icon from old HoG icons or can't reuse already existing sprites
2. Replace some input thingies with radial menu's

**Comment by Bibby0110** [comment]
@JamieD1

**Comment by Bibby0110** [comment]
Conflicts + we don't believe the gamemode fits in with the current desired player interactions, sorry.

**Pull request closed without merging**